server/websocket_fallback.py

```
@@ -156,81 +184,145 @@ def _handle_client(self, client, addr, handler):
             logger.error(f"Client error from {addr}: {e}")
         finally:
             client.close()
             logger.info(f"Connection closed for {addr}")
-
-    def start(self, handler):
+            if "send_func" in locals() and disconnect_handler:
+                try:
+                    res = disconnect_handler(send_func)
+                    if inspect.isawaitable(res):
+                        loop.run_until_complete(res)
+                except Exception as e:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)